source/draw/RasterBgr.ooc

```diff
@@ -111,18 +111,23 @@ RasterBgr: class extends RasterPacked {
 		This new(ByteBuffer new(data as UInt8*, x * y * requiredComponents), IntSize2D new(x, y))
 	}
 	convertFrom: static func (original: RasterImage) -> This {
-		result := This new(original size)
-		row := result buffer pointer as Long
-		rowLength := result size width
-		rowEnd := row as ColorBgr* + rowLength
-		destination := row as ColorBgr*
-		f := func (color: ColorBgr) {
-			(destination as ColorBgr*)@ = color
-			destination += 1
-			if (destination >= rowEnd) {
-				row += result stride
-				destination = row as ColorBgr*
-				rowEnd = row as ColorBgr* + rowLength
+		result: This
+		if (original instanceOf?(This))
+			result = (original as This) copy()
+		else {
+			result = This new(original size)
+			row := result buffer pointer as Long
+			rowLength := result size width
+			rowEnd := row as ColorBgr* + rowLength
+			destination := row as ColorBgr*
+			f := func (color: ColorBgr) {
+				(destination as ColorBgr*)@ = color
+				destination += 1
+				if (destination >= rowEnd) {
+					row += result stride
+					destination = row as ColorBgr*
+					rowEnd = row as ColorBgr* + rowLength
+				}
 			}
 		}
 		original apply(f)
```

Comment:
This needs to be moved inside the `else` clause or `f` will be out of scope when you call `original appl(f)`.